packages/instant/src/util/heartbeater.ts

```diff
@@ -0,0 +1,49 @@
+import * as _ from 'lodash';
+
+type HeartbeatableFunction = () => Promise<void>;
+export class Heartbeater {
+    private _intervalId?: number;
+    private _hasPendingRequest: boolean;
+    private readonly _performImmediatelyOnStart: boolean;
+    private readonly _performFunction: HeartbeatableFunction;
+
+    public constructor(performingFunctionAsync: HeartbeatableFunction, performImmediatelyOnStart: boolean) {
+        this._performFunction = performingFunctionAsync;
+        this._hasPendingRequest = false;
```

Comment:
we have `intervalUtils.setAsyncExcludingInterval` in `@0x/utils` that takes care of this concurrent request handling

Reply:
Ah this is nice!  Wish I knew this before writing up my own though.  When possible, we should try to bring up any knowledge like this when "grooming" stories